### PR TITLE
feat(services/markdown): add scroll to element

### PR DESCRIPTION
**Pull request by danivm**
Add scroll to an element after markdown is rendered.

[SCMI-21452](https://jira.scmspain.com/browse/SCMI-21452)

### CODE REVIEW

**Review by davecarter — APPROVED** [review]

**Review by midudev — APPROVED** [review]

File: components/services/markdown/src/index.js [hunk 1]
```diff
@@ -28,6 +28,14 @@ class ServiceMarkdown extends Component {
     )
   }
 
+  componentDidUpdate() {
+    const id = document.location.hash.substring(1)
```
Comment (jcger) [inline components/services/markdown/src/index.js:32]: I think this makes the component less reusable, "id" should be a prop passed by to the dumb UI component and the logic behind loading this id should be done in the wrapping component from each vertical

**Review by jcger — CHANGES_REQUESTED** [review]

**Review by jcger — APPROVED** [review]

File: components/services/markdown/src/index.js [hunk 2]
```diff
@@ -28,6 +28,14 @@ class ServiceMarkdown extends Component {
...
+    if (id) {
+      const element = document.getElementById(id)
+      element && element.scrollIntoView({block: 'start', behavior: 'smooth'})
```
Comment (davidbarna) [inline components/services/markdown/src/index.js:35]: FYI: smooth not supported by IE11

File: components/services/markdown/src/index.js [hunk 3]
```diff
@@ -28,6 +28,14 @@ class ServiceMarkdown extends Component {
     )
   }
 
+  componentDidUpdate() {
```
Comment (davidbarna) [inline components/services/markdown/src/index.js:31]: componentDidUpdate will be called every time the parent component is rendered, even if your props are unchanged. So, you will rescroll over and over on each update. You must control that.
The scroll should be done only html changes.
So you can fix this with a shouldComponentUpdate.

File: components/services/markdown/src/index.js (hunk 1 above)
Comment (midudev) [reply components/services/markdown/src/index.js:32]: We've been talking about this and we're keeping as it is to keep the native spec of HTML with hash and id.

Comment (davidbarna) [reply components/services/markdown/src/index.js:32]: We are generating HTML anyways in a web environment in a cycle method that only executes in browser. So it's ok to me to do it here.

File: components/services/markdown/src/index.js (hunk 2 above)
Comment (midudev) [reply components/services/markdown/src/index.js:35]: I think he's aware about that. The good thing is that the same code works in IE11 but without behaviour.

File: components/services/markdown/src/index.js [hunk 4]
```diff
@@ -28,6 +28,18 @@ class ServiceMarkdown extends Component {
...
+    const id = document.location.hash.substring(1)
+    if (id) {
+      const element = document.getElementById(id)
+      element && element.scrollIntoView({block: 'start', behavior: 'smooth'})
+    }
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return nextState.html !== this.state.html
```
Comment (midudev) [inline components/services/markdown/src/index.js]: I think we could simplify this kind of comparison. Instead comparing the huge html, creating a new state and using it in the shouldComponentUpdate to only do the action once.

**Review by carlosvillu — APPROVED** [review]

**Pull request merged**